File: Day03/PerfectNo.py
class Solution(object):
    def checkPerfectNumber(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num <= 1:
            return False
            
        s = 1  
        n = int(num**0.5) + 1
        
        for i in range(2, n):
            if num % i == 0:
                s += i
                c = num // i
                if c != i:
                    s += c
                if s > num:
                    return False
        
        return s == num

def main():
    sol = Solution()
    tests = [6, 28, 496, 8128, 5, 12, 33550336]
    for n in tests:
        res = sol.checkPerfectNumber(n)
        print(f"{n}: {'Perfect' if res else 'Not perfect'}")

if __name__ == "__main__":
    main()

File: Day03/PerfectNo.java
public class PerfectNo {
    public boolean checkPerfectNumber(int num) {
        if (num <= 1) {
            return false;
        }
        
        int sum = 1; 
        int sqrt = (int)Math.sqrt(num) + 1;
        
        for (int i = 2; i < sqrt; i++) {
            if (num % i == 0) {
                sum += i;
                int comp = num / i;
                if (comp != i) {
                    sum += comp;
                }
                if (sum > num) {
                    return false;
                }
            }
        }
        
        return sum == num;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        int[] tests = {6, 28, 496, 8128, 5, 12, 33550336};
        
        for (int n : tests) {
            boolean res = sol.checkPerfectNumber(n);
            System.out.println(n + ": " + (res ? "Perfect" : "Not perfect"));
        }
    }
}